### PR TITLE
Fix new joiner infinite loop 

### PR DESCRIPTION
**Pull request by nickoferrall**
This PR _may_ resolve issue #4542. 

[This](https://sentry.io/organizations/parabol/issues/1641789488/?project=107196&query=level%3Afatal&statsPeriod=14d) and [this](https://sentry.io/organizations/parabol/issues/2109105416/?project=107196&query=level%3Afatal&statsPeriod=14d) Sentry issue suggest that the new joiner can infinitely bounce between the [ViewerNotOnTeam](https://github.com/ParabolInc/parabol/blob/91aff6e5c28bf344521db5701b0113401a2bc1dc/packages/client/components/ViewerNotOnTeam.tsx#L44) and the [MeetingSelector](https://github.com/ParabolInc/parabol/blob/01bfad9f162ef8f69c034a956bfbce9db3b1e9b9/packages/client/components/MeetingSelector.tsx#L35) components.

I haven't been able to reproduce the issue but I suspect that the `teamId` may be added to Relay's `authObj` before the server updates with the meeting. `isOnTeam` is therefore true in `ViewerNotOnTeam` but the `MeetingSelector` sends the viewer back to `ViewerNotOnTeam` as it doesn't have the `meeting` yet.

### CODE REVIEW

**Comment by mattkrick** [comment]
these are always fun ones. if your assumption is correct, then we should just ignore what the authObj says. i'll add a couple comments

File: packages/client/components/ViewerNotOnTeam.tsx [hunk 1]
```diff
@@ -41,10 +42,15 @@ const ViewerNotOnTeam = (props: Props) => {
...
-        const redirectTo = getValidRedirectParam()
-        const nextRoute = redirectTo || `/team/${teamId}`
+        return
+      }
+      const redirectTo = getValidRedirectParam()
+      // if already on the team, go to team dash
+      const nextRoute = redirectTo || `/team/${teamId}`
+      const isRoutingToMeeting = nextRoute === `/meet/${meetingId}`
+      const isRoutingToTeam = nextRoute === `/team/${teamId}`
+      const isNextRoutePossible = (isOnTeam && isRoutingToTeam) || (meeting && isRoutingToMeeting)
```
Comment (mattkrick) [inline packages/client/components/ViewerNotOnTeam.tsx:52]: if the problem is with the auth token updating before the query updates, I think we can simply this:
```
if (teamInvitation) acceptTeamInvitation...
else if (meeting) history.replace(/meet/meetingId)
else if (team) history.replace('/team/teamId')
else PushInvitation

Comment (nickoferrall) [reply packages/client/components/ViewerNotOnTeam.tsx]: makes sense! I've removed the `authObj` check here and in the TeamContainer

**Comment by mattkrick** [comment]
TESTS
- [ ] invite user b to the team. User b goes to the meeting route. the team invitation is auto accepted & they go to the meeting
- [ ] invite user b to the team. User b goes to the team route. the team invitation is auto accepted & they go to the team dash
- [ ] while user b is on the invitation-required route, user a invites user b from a meeting via email. they get redirected to the meeting
- [ ] same as above, but invite comes from the team dash. user b lands on the team dash
- [ ] user b goes to team dash without an invite. they go to invitation-required & user a gets a snackbar asking to accept
- [ ] user b goes to meeting without an invite. they go to invitatino-required & user a gets a snackbar asking to accept
- [ ] user b is on the team & goes to the meeting route, they go to the meeting

**Pull request merged**